programs/psa/key_ladder_demo.c

```diff
@@ -58,6 +58,7 @@
 
 #include "mbedtls/platform.h" // for mbedtls_setbuf
 #include "mbedtls/platform_util.h" // for mbedtls_platform_zeroize
+#include "mbedtls/md.h"
```

Comment:
Hum, considering we're testing for `PSA_WANT_ALG_SHA256` below, shouldn't this be `psa/crypto.h` instead?

Reply:
I don't know, now I think it looks weird with the comment mentioning this is for MD_CAN_XXX macros but we're not using any in this file.

Reply:
Fixed :)

Reply:
PSA example programs should avoid including non-PSA headers: they're supposed to work with any PSA implementation. They should definitely not include cryptography-related mbedtls headers such as `mbedtls/md.h`. They might include platform headers (and even that is debatable).